Remove entwine from the scheduler

Arnei opened the pull request:
Replace entwine code with plain Java in the scheduler-api, scheduler-impl and scheduler-remote modules.

Should result in no functional changes.

### How to test this patch

Schedule events as you would normally.

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://docs.opencast.org/develop/developer/#participate/development-process/#automatically-closing-issues-when-a-pr-is-merged) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)
* [ ] explain why it needs to be merged into the legacy branch, if it is targeting the legacy branch

Arnei committed:
Remove entwine from the scheduler

Replace entwine code with plain Java in the scheduler-api,
scheduler-impl and scheduler-remote modules.

gregorydlogan reviewed (changes requested):
I'm kinda shocked at this passes the tests.  Locally I needed
```
diff --git a/modules/scheduler-impl/pom.xml b/modules/scheduler-impl/pom.xml
index 24971cc404..117891b676 100644
--- a/modules/scheduler-impl/pom.xml
+++ b/modules/scheduler-impl/pom.xml
@@ -79,6 +79,7 @@
     <dependency>
       <groupId>com.entwinemedia.common</groupId>
       <artifactId>functional</artifactId>
+      <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>joda-time</groupId>
```

To pass the dependency checker

Arnei committed:
Add scope to entwine dependency in scheduler-impl

Not having this scope apparently causes build failures
on some systems for some reason.

Arnei commented:
Builds for me locally without scope, but also with scope, so I added it. *shrug*

gregorydlogan commented on modules/index-service/src/test/java/org/opencastproject/index/service/impl/IndexServiceImplTest.java:
Line 86 still has an import for `Opt`

gregorydlogan commented on modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/impl/IngestServiceImpl.java:
Line 102 has an `Opt` import still, and once that's removed then the pom file needs the whole `functional` library removed.

gregorydlogan commented on modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/impl/SchedulerUtil.java:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:compile (default-compile) on project opencast-scheduler-impl: Compilation failure: Compilation failure: 
[ERROR] /home/greg/opencast/upstream/modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/impl/SchedulerUtil.java:[226,60] cannot access com.entwinemedia.fn.data.Opt
[ERROR]   class file for com.entwinemedia.fn.data.Opt not found
[ERROR] /home/greg/opencast/upstream/modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/impl/SchedulerUtil.java:[234,56] for-each not applicable to expression type
[ERROR]   required: array or java.lang.Iterable
[ERROR]   found:    com.entwinemedia.fn.data.Opt
[ERROR] /home/greg/opencast/upstream/modules/scheduler-impl/src/main/java/org/opencastproject/scheduler/impl/SchedulerServiceImpl.java:[565,19] cannot access com.entwinemedia.fn.Stream
[ERROR]   class file for com.entwinemedia.fn.Stream not found
[ERROR] -> [Help 1]
```

which is what the unit tests are spitting out.

gregorydlogan reviewed (changes requested):
Found a few initial things.  Gave up at `SchedulerUtil`.

Arnei committed:
Merge branch 'develop' into remove-entwine-from-scheduler

Arnei committed:
Remove unused imports and fix dependencies

Arnei commented:
Hopefully this resolves all the things you found.

gregorydlogan commented:
Yep, works for me!
